subprojects/data-structures/src/main/java/org/gradle/internal/Try.java

```diff
@@ -221,7 +223,19 @@ public boolean isSuccessful() {
 
         @Override
         public T get() {
-            throw UncheckedException.throwAsUncheckedException(failure);
+            if (failure instanceof InterruptedException) {
+                Thread.currentThread().interrupt();
+            }
+            if (failure instanceof RuntimeException) {
+                throw (RuntimeException) failure;
```

Comment:
I wonder if it wouldn't be better to move `Cast` and `UncheckedException` to `base-annotations` (and rename base-annotations` to something else. So we could have all this util-code in one place re-usable instead of duplicating this non-trivial amount of code here.

Reply:
I think we should do things like this once we have a way to merge small JARs.

Reply:
Why not move it now to `base-annotations`? We have the small jar already.

Reply:
Because they are not annotations. I'd rather fix this properly with a new subproject, and until then I think it's fine to do the changes above. It's not some super-complicated logic that we are copying here.